### PR TITLE
Update friends controller to have create method

### LINKED ISSUE #11: 3B Dashboard: Friends
As an authenticated user,
I see a section for friends,
In this section, there should be a text field to enter a friend's email and a button to "Add Friend"

**Scenarios**: 
1. If I have not added any friends there should be a message. "You currently have no friends".

2. If I have added friends, I should see a list of all my friends.

**Details:** Users should be able to add a friend by their email address, as long as, the friend is a user of our application and exists in our database. 

**Example:** 
Bugs Bunny and Lola Bunny are users of our application, but Daffy Duck is not. 
1. When Bugs Bunny enters `lola_bunny@gmail.com` to add friend it will be successful and Lola should show up as Bugs Bunny's friend.
2. When Bugs Bunny enters `daffy_duck@gmail.com` to add friend it should give an error message that the user does not exist. 

**Tips:** You'll want to research self-referential `has_many through`. [Here](http://blog.hasmanythrough.com/2007/10/30/self-referential-has-many-through) is a good starting point to understand the concept. You will probably need to do more googling but that's part of the fun ;)

- [ ] Write a happy path test 
- [ ] Write a sad path test

### PR DESCRIPTION
**Pull request by catherinemdean15**
Add flash messages for successful and unsuccessful creation
Add search feature for friends
Add dashboard tests to list and add friends

(blank line)
Co-authored-by: Joe Jiang <69106714+ninesky00@users.noreply.github.com>
Co-authored-by: Cydnee Owens <67389821+cowens87@users.noreply.github.com>

### CODE REVIEW

File: app/controllers/friends_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      redirect_to dashboard_user_path(current_user)
+    end
+  end
+
+  private
+
+  def friend_params
+    params.require(:friend).permit(:email)
+  end
+end
```
Comment (ninesky00) [inline app/controllers/friends_controller.rb:19]: is the strong params necessary?

File: app/views/users/dashboard.html.erb [hunk 2]
```diff
@@ -4,10 +4,20 @@
 <%= button_to 'Search for movies', movies_path, method: :get %></li>
 
 <section id="friends">
-  <% current_user.buddies.each do |buddy|%>
-    <% friend = User.find(buddy.friend_id) %>
-    <%= friend.email %>
-  <% end %>
+  <h3>Friends</h3>
+    <% if current_user.followings == [] %>
+      <%= "You currently have no friends" %>
```
Comment (ninesky00) [inline app/views/users/dashboard.html.erb]: This might touch on MVC principles

File: app/controllers/friends_controller.rb (hunk 1 above)
Comment (catherinemdean15) [reply app/controllers/friends_controller.rb:19]: great question! I'm not totally sure. We can circle back!

File: app/views/users/dashboard.html.erb (hunk 2 above)
Comment (catherinemdean15) [reply app/views/users/dashboard.html.erb]: Great catch!

**Pull request merged**